pkgs/applications/editors/neovim/default.nix

```
@@ -19,6 +20,16 @@ let
         nvim-client luv coxpcall busted luafilesystem penlight inspect
       ]
     ));
+  codegenLua =
+    if lua.pkgs.isLuaJIT
+      then
+        let deterministicLuajit =
+          lua.override {
+            deterministicStringIds = true;
+            self = deterministicLuajit;
+          };
+        in deterministicLuajit.withPackages(ps: [ ps.mpack ps.lpeg ])
+      else lua;
```

```suggestion
  codegenLua = let
    deterministicLuajit = lua.override {
      deterministicStringIds = true;
      self = deterministicLuajit;
     };
  in
    if lua.pkgs.isLuaJIT then
      deterministicLuajit.withPackages (ps: with ps; [ mpack lpeg ])
    else
      lua;
```
a tiny reformat can make wonders for readability

---

I don't agree this improves readability, as it increases the scope of the `deterministicLuajit` variable outside of the `lua.pkgs.isLuaJIT` condition.

Even if nix is lazy, so it doesn't matter runtime-wise, this hurts readability: now someone reading this code for a use case where `lua` is the regular lua and not luajit will be distracted by the introduction of a `deterministicLuajit` passing nonexistent parameter to the lua derivation, which then isn't used.

---

I guess I never look at such things or it doesn't bother me.